### DISCUSSION TITLE
How to use monai inference service in Windows 10?

### DISCUSSION THREAD

**Discussion by ty010624** [Project-MONAI/monai-deploy-app-sdk#324] [General]
Hi,

I'm trying to follow the "05_full_tutorial" notebook but not able to mount a local folder from Windows 10 host to the kubernetes pod. I'm using Docker Desktop v4.11.0 with Kubernetes v1.24.2. I do not use GPU for inferencing (gpu=0). 

I tried to set my hostVolumePath to: /run/desktop/mnt/host/c/monai/payload

I was able to start up the monai inference service. The input was uploaded to the /input folder. But I wasn't able to get any output in the /output folder. There was an error in the pod:

ERROR:root:
Traceback (most recent call last):
  File "/monai_inference/monaiinference/main.py", line 147, in upload_file
    raise HTTPException(status_code=500, detail="Request failed since MAP container's pod failed")
fastapi.exceptions.HTTPException

Not sure what was the problem...

Thanks,
Kelly

**Comment by MMelQin** [comment]
@ty010624 Thanks for trying out the MONAI Deploy and specifically the App SDK.

I'll try to identify a possible root cause of the MAP container pod failure. If possible, can you check the version of monai package in the MAP container image? The App SDK v0.4 was released on the heal of monai v0.9.0 and tested for compatibility, however, monnai v0.9.1 was released soon after on July 25, 2022, and the MetaTensor introduced in this version caused compatibility issues with App SDK v0.4. The inference module in the App SDK and in the tutorial did not pin the monai version, using monai>=0.9.0 or similar, so the latest monai is installed when the MPA docker image is built, so at runtime, it causes inference execution to fail, hence the container failure. The A patch build has been planned to update the inference code to be compatible with MetaTensor in monai>=v0.9.1.

In the meantime, can you try to pin the monai version in the tutorial, replacing the multiple occurrences of `monai>=0.8.1` with `monai==0.8.1`, or even `monai==0.9.0`, and try again?

**Comment by ty010624** [comment]
I have fixed the issue. It was mainly due to some wrong configuration on my end. Now the inference service is working fine on my Windows 10 machine. Thank you very much for your reply!

Yes, I did notice the incompatibility issue with latest monai. I had fixed monai to 0.8.1 in the image I created. The issue was on my list to investigate further. But since I know it used to work, I just reverted to previous working version and carried on. But thank you very much for the detailed explanation! Now I know what was the root cause of it.

**Reply by MMelQin** [reply]
Thanks @ty010624. Great to hear the issues had already been resolved, and thanks again for test driving MONAI and MONAI Deploy. We are planning to release either App SDK v0.4 patch or v0.5 within 2-3 weeks, in time for MICCAI 2022 (Sept 18th).
